smarts/core/sumo_road_network.py

```diff
@@ -121,6 +125,30 @@ def _compute_road_polygons(self, scale):
 
         return polys
 
+    def _snap_internal_edges(self, lane_to_poly, snap_threshold=2):
+        # HACK: Internal edges that have tight curves, when buffered their ends do not
```

Comment:
I am not sure this is a hack since the underlying data is not conducive to fitted segments.

Reply:
It in fact is, I can explain more this evening.